flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointRecoveryFactory.java

```diff
@@ -31,10 +35,17 @@
      * @param jobId Job ID to recover checkpoints for
      * @param maxNumberOfCheckpointsToRetain Maximum number of checkpoints to retain
      * @param userClassLoader User code class loader of the job
+     * @param sharedStateRegistryFactory Simple factory to produce {@link SharedStateRegistry}
+     *     objects.
+     * @param ioExecutor Executor used to run (async) deletes.
      * @return {@link CompletedCheckpointStore} instance for the job
      */
     CompletedCheckpointStore createRecoveredCompletedCheckpointStore(
-            JobID jobId, int maxNumberOfCheckpointsToRetain, ClassLoader userClassLoader)
+            JobID jobId,
+            int maxNumberOfCheckpointsToRetain,
+            ClassLoader userClassLoader,
+            SharedStateRegistryFactory sharedStateRegistryFactory,
```

Comment:
I am wondering if we need the `SharedStateRegistryFactory` parameter. I'd say it should be up to the `CompletedCheckpointStore`/`CheckpointRecoveryFactory` to decide about the implementation.